SWDEV-552580 - Update instructions to build HIP from source in monorepo (#806)
ROCm/rocm-systems: projects/hip/docs/install/build.rst

Reviewer: The text on line 81 says: 
Note, if you don't specify ``CMAKE_INSTALL_PREFIX``, the HIP runtime is installed at ``<ROCM_PATH>``.

Can you say where ROCM_PATH is by default?

Reviewer: The text on line 84 says:
By default, release version of HIP is built. If need debug version, you can put the option ``CMAKE_BUILD_TYPE=Debug`` in the command line.

Change to:
By default, the release version of HIP is built. If you need a debug version, you can put the option ``CMAKE_BUILD_TYPE=Debug`` in the command line.

Reviewer: On line 89 it looks like there is orphan text that should be removed:
"                          environment variable."

Building the HIP runtime
==========================================================

> Starting in ROCM 7.1 release, HIP will be a part of core rocm projects resides in a monorepository.

Reviewer: ```suggestion
In the ROCM 7.1 release, HIP is a part of core ROCm projects that reside in the ``rocm-systems`` monorepository.
```

Reviewer: This seems like an important note, as 7.1 is the first time users will be seeing the rocm-systems mono repo, but it lacks further information. Maybe add some stuff about the monorepo, how it works, why it is used, and refer to the README.md file in the rocm-systems monorepo for more information?

   export ROCM_BRANCH=rocm-6.1.x

> Starting in ROCM 7.1 release, HIP will be a part of core rocm projects resides in a mono-repository,

Reviewer: I don't understand the need for the rocm-6.1.x branch above if this is for the 7.1 version of HIP. It seems like the instructions would be for the current release, and prior instructions could be accessed from prior releases?

Author: Agree. Thanks. Updated.

      #. Get HIP source code.

         .. note::

>             Starting in ROCM 7.1, a new monorepository is used to get HIP source codes from.

Reviewer: This note seems the same as the earlier text on line 40. I think you only need one of these, but it is not clear to me at least where the note would best be placed.

         .. code-block:: shell

            git clone -b "$ROCM_BRANCH" git@github.com:ROCm/rocm-systems.git   

      #. Set the environment variables.

         .. code-block:: shell

            export CLR_DIR="$(readlink -f rocm-systems/projects/clr)"
            export HIP_DIR="$(readlink -f rocm-systems/projects/hip)"

      #. Build HIP.

         .. code-block:: shell

            cd "$CLR_DIR"
            mkdir -p build; cd build
            cmake -DHIP_COMMON_DIR=$HIP_DIR -DHIP_PLATFORM=amd -DCMAKE_PREFIX_PATH="/opt/rocm/" -DCMAKE_INSTALL_PREFIX=$PWD/install -DCLR_BUILD_HIP=ON -DCLR_BUILD_OCL=OFF ..
            make -j$(nproc)
            sudo make install

         .. note::

            If ``CMAKE_INSTALL_PREFIX`` is not explicitly specified, the HIP runtime will be installed at
            ``<ROCM_PATH>``, which is by default at the path ``/opt/rocm``.

            By default, the release version of HIP is built. If you need a debug version, you can put the option ``CMAKE_BUILD_TYPE=Debug`` in the command line.

         Default paths:

            * HIP is installed into ``<ROCM_PATH>``. This can be overridden by setting the ``INSTALL_PREFIX`` as the command option.

>                environment variable.

Reviewer: This looks like orphan text?

         OpenCL provides headers that ROCclr runtime currently depends on.
         hipother provides headers and implementation specifically for non-AMD HIP platforms, like NVIDIA.

         .. note::

>             Starting in ROCM 7.0, a new mono-repository is used to get HIP source codes from.

Reviewer 2: have indentation similar to line 69.

Author: Updated. Thanks.

         .. code-block:: shell

            git clone -b "$ROCM_BRANCH" git@github.com:ROCm/rocm-systems.git   

      #. Set the environment variables.
         
         Before ROCM 7.0:

         .. code-block:: shell

            export CLR_DIR="$(readlink -f clr)"
            export HIP_DIR="$(readlink -f hip)"

>          Staring in ROCM 7.0:

Reviewer 3: typo, starting

Author: Updated. Thanks.

         * Get HIP tests source code.

           Before ROCM 7.0:

            .. code-block:: shell

               git clone -b "$ROCM_BRANCH" https://github.com/ROCm/hip-tests.git
               export HIPTESTS_DIR="$(readlink -f hip-tests)"

>            Start in ROCM 7.0:

Reviewer 3: Starting

Author: Updated. Thanks.
